app/lib/summary_stats_utils.rb

```diff
@@ -0,0 +1,76 @@
+##
+# PortalUtils: generic class with server stats/maintenance methods
+#
+# note: all instances of :start_date and :cutoff_date are inclusive
+##
+class SummaryStatsUtils
+
+  # get a snapshot of user counts/activity up to a given date
+  # will give count of users as of that date, and number of active users on that date
+  def self.total_and_active_user_counts(cutoff_date: Time.zone.today)
+    # make sure to make cutoff_date one day forward to include any users that were created on cutoff date
+    next_day = cutoff_date + 1.day
+    total_users = User.where(:created_at.lte => next_day).count
+    active_users = User.where(:current_sign_in_at => (cutoff_date..next_day)).count
+    {total: total_users, active: active_users}
+  end
+
+  # get a count of all submissions launch from the portal in a given 2 week period
+  # defaults to a time period of the last two weeks from right now
+  def self.analysis_submission_count(start_date: DateTime.now - 2.weeks, cutoff_date: DateTime.now)
+    AnalysisSubmission.where(:submitted_on => (start_date..cutoff_date), submitted_from_portal: true).count
+  end
+
+  # get a count of all studies created on the requested day
+  def self.study_creation_count(cutoff_date: Time.zone.today)
+    Study.where(:created_at => (cutoff_date..(cutoff_date + 1.day))).count
+  end
+
+  # perform a sanity check to look for any missing files in remote storage
+  # returns a list of all missing files for entire portal for use in nightly_server_report
+  def self.storage_sanity_check
+    missing_files = []
+    Study.where(queued_for_deletion: false, detached: false).each do |study|
+      begin
+        study_missing = study.verify_all_remotes
+        study_missing.any? ? missing_files += study_missing : nil
+      rescue => e
+        # check if the bucket or the workspace is missing and mark study accordingly
+        study.set_study_detached_state(e)
+        ErrorTracker.report_exception(e, nil, {})
+        Rails.logger.error  "Error in retrieving remotes for #{study.name}: #{e.message}"
+        missing_files << {filename: 'N/A', study: study.name, owner: study.user.email, reason: "Error retrieving remotes: #{e.message}"}
+      end
+    end
+    missing_files
+  end
+
+  # disk usage stats
+  def self.disk_usage
+    disk = `df -h /home/app/webapp`.split("\n").last
```

Comment:
I recall historic security concerns with system calls in Rails.  Is the principle for this being OK that it accepts no user input, and is read only?

Reply:
To be honest, I'm not psyched about this, but I don't know of another way to do it in Ruby.  In theory the concern is warranted but acceptable given the two conditions you named.

I'm going to take a quick pass through the interwebs to see if there's a gem that can safely do this when I address the other PR concerns.